rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/SelectionManager.java

```diff
@@ -587,38 +593,52 @@ public String toString() {
                     return "executing script on " + node.getNodeURL();
                 }
             });
+            scriptHosts.add(node.getHostName());
         }
 
         // Invoke all Callables and get the list of futures
-        List<Future<ScriptResult<T>>> futures = null;
+        List<Future<ScriptResult<T>>> futures = new LinkedList<>();
         try {
-            futures = this.scriptExecutorThreadPool.invokeAll(scriptExecutors, timeout, TimeUnit.MILLISECONDS);
+            futures = this.scriptExecutorThreadPool.invokeAll(scriptExecutors,
+                                                              allScriptExecutionsTimeout,
+                                                              TimeUnit.MILLISECONDS);
         } catch (InterruptedException e) {
             logger.warn("Interrupted while waiting, unable to execute all scripts", e);
             Thread.currentThread().interrupt();
         }
 
         final List<ScriptResult<T>> results = new LinkedList<>();
+        List<RMNode> nodesList = new ArrayList<>(nodes);
 
         int index = 0;
-        // waiting for the results
         for (final Future<ScriptResult<T>> future : futures) {
-            final String description = scriptExecutors.get(index++).toString();
-            ScriptResult<T> result = null;
+            final String description = scriptExecutors.get(index).toString();
+            String nodeURL = nodesList.get(index).getNodeURL();
+
+            ScriptResult<T> result;
             try {
+                logger.info("Awaiting script result on " + nodeURL);
```

Comment:
As we have two nested future awaiting (the ProActive future a bit higher and the Java future here) I thought it would be good to track the progress of both.